js/gemini.js

```diff
@@ -991,6 +1006,88 @@ module.exports = class gemini extends Exchange {
+        params = this.omit (params, 'type');
+        const rawStopPrice = this.safeString2 (params, 'stop_price', 'stopPrice');
+        params = this.omit (params, [ 'stop_price', 'stopPrice', 'type' ]);
+        if (type === 'stopLimit') {
+            throw new ArgumentsRequired (this.id + ' createOrder () requires a stopPrice parameter or a stop_price parameter for ' + type + ' orders');
+        }
+        if (rawStopPrice !== undefined) {
+            request['stop_price'] = this.priceToPrecision (symbol, rawStopPrice);
+            request['type'] = 'exchange stop limit';
+        } else {
+            // No options can be applied to stop-limit orders at this time.
+            const timeInForce = this.safeString (params, 'timeInForce');
+            params = this.omit (params, 'timeInForce');
+            if (timeInForce !== undefined) {
+                if ((timeInForce === 'IOC') || (timeInForce === 'immediate-or-cancel')) {
+                    request['options'] = [ 'immediate-or-cancel' ];
+                } else if ((timeInForce === 'FOK') || (timeInForce === 'fill-or-kill')) {
+                    request['options'] = [ 'fill-or-kill' ];
+                } else if (timeInForce === 'PO') {
+                    request['options'] = [ 'maker-or-cancel' ];
+                }
+            }
+            const postOnly = this.safeValue (params, 'postOnly', false);
+            params = this.omit (params, 'postOnly');
+            if (postOnly) {
+                request['options'] = [ 'maker-or-cancel' ];
+            }
+            // allowing override for auction-only and indication-of-interest order options
+            const options = this.safeString (params, 'options');
+            if (options !== undefined) {
+                request['options'] = [ options ];
+            }
+        }
+        const response = await this.privatePostV1OrderNew (this.extend (request, params));
+        //
+        //      {
+        //          "order_id":"106027397702",
+        //          "id":"106027397702",
+        //          "symbol":"etheur",
+        //          "exchange":"gemini",
+        //          "avg_execution_price":"2877.48",
+        //          "side":"sell",
+        //          "type":"exchange limit",
+        //          "timestamp":"1650398122",
+        //          "timestampms":1650398122308,
+        //          "is_live":false,
+        //          "is_cancelled":false,
+        //          "is_hidden":false,
+        //          "was_forced":false,
+        //          "executed_amount":"0.014434",
+        //          "client_order_id":"1650398121695",
+        //          "options":[],
+        //          "price":"2800.00",
+        //          "original_amount":"0.014434",
+        //          "remaining_amount":"0"
+        //      }
+        //
+        return this.parseOrder (response);
+    }
+
+    async createOrder2 (symbol, type, side, amount, price = undefined, params = {}) {
```

Comment:
We will probably need to remove createOrder2, I assume it's a leftover after testing.

Reply:
I am so sorry, good spot, thanks